allennlp/data/iterators/transform_iterator.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from typing import Dict, Union, Iterable, Iterator, List
+from collections import defaultdict
+import itertools
+import math
+
+import torch
+from torch.utils.data import DataLoader
+
+
+from allennlp.common.util import is_lazy, ensure_list
+from allennlp.data.iterators.data_iterator import DataIterator
+from allennlp.data.dataset import Batch
+from allennlp.data.instance import Instance
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data import transforms
+
+logger = logging.getLogger(__name__)
+
+TensorDict = Dict[str, Union[torch.Tensor, Dict[str, torch.Tensor]]]
+
+
+@DataIterator.register("transform")
+class TransformIterator:
```

Comment:
Yeah... It can subclass it, but the interface for the private `_create_batches` method is different. The objective is to remove all the other iterators anyway, so idk what's best/if this distinction is important.